acl_linux.go

```diff
@@ -37,36 +37,89 @@ const (
 	aclUndefinedID = math.MaxUint32 // defined in sys/acl.h
 )
 
+const defaultbuflen = 64
+
+type file interface {
+	Getxattr(attr string, dest []byte) (int, error)
+	Setxattr(attr string, dest []byte, flags int) error
+	Stat() (os.FileInfo, error)
+}
+
+type Path string
+
+func (p Path) Getxattr(attr string, dest []byte) (sz int, err error) {
+	if sz, err = syscall.Getxattr(string(p), attr, dest); sz == -1 && err == syscall.ERANGE {
+		if sz, err = syscall.Getxattr(string(p), attr, nil); sz <= 0 {
+			return
+		}
+		dest = make([]byte, sz)
+		sz, err = syscall.Getxattr(string(p), attr, dest)
```

Comment:
Won't doing it this way only store the results temporarily inside the function? When you do `dest = make([]byte, sz)`, that will cause `dest` to point to a temporary allocation which isn't visible from the caller, so all they'll see is that the slice they passed hasn't been modified. I think you should just have this call `syscall.Getxattr` and nothing else and put this error handling in the caller. The same goes for the method on `File`.

Reply:
I agree, my mistake